tla-pp/src/test/scala/at/forsyte/apalache/tla/pp/TestConstAndDefRewriter.scala

```diff
@@ -105,8 +108,18 @@ class TestConstAndDefRewriter extends AnyFunSuite with BeforeAndAfterEach {
     val rewritten = new ConstAndDefRewriter(new IdleTracker())(root)
     assert(rewritten.constDeclarations.isEmpty)
     assert(rewritten.operDeclarations.size == 1)
-    val expected =
-      TlaOperDecl("BoolMin", List(OperParam("S")), tla.choose(tla.name("x"), tla.name("S"), tla.bool(true)))
+    val b = BoolT1()
+    val expected: TlaOperDecl =
+      tla.declWithInferredParameterTypes(
+          "BoolMin",
+          tla.choose(
+              tla.name("x", b),
+              tla.name("S", SetT1(b)),
```

Comment:
I usually define the parser in tests:

```scala
  private val parser = DefaultType1Parser
```

Then you can conveniently write `parser("Set(b)")` instead of directly constructing type expressions.

Reply:
Not sure `parser("Set(b)")` is more convenient than `SetT1(b)`

Reply:
In this case, it is not easier. It is usually easier to read for complex types like `(Int, Str) => (Int -> Str)`, as the type grammar is more compact.

Reply:
Ok, I can add an implicit `Str => TlaType1` method in `convenience` that calls the parser. Then you can write `name("x", "a -> b")` directly.

Reply:
Please don't. Implicit conversion from strings to types is a really bad idea. How about adding a convenience method to the builder, e.g., `type("a -> b")`?

Reply:
The type parser should ignore whitespaces, as every other parser does.